aux/src/FrameSync.cxx

```diff
@@ -62,8 +57,20 @@ bool Aux::FrameSync::operator()(input_queues& iqs, output_queues& oqs)
         return (*this)(iqs, oqs);
     }
 
-    // If we reach here then we do not have fully populated input
-    // streams and must wait for more.
-    
+    // push eos
+    for (size_t ind=0; ind<nin; ++ind) {
+        auto& iq = iqs[ind];
+        if (iq.empty()) {
+            continue;
+        }
+        auto frame = iq.front();
+        if (!frame) {
+            std::get<0>(oqs).push_back(nullptr); // forward EOS
+            iq.pop_front();
+            return (*this)(iqs, oqs);
+        }
+    }
+
+    /// FIXEME: check all empty
```

Comment:
This might be correct but I'm having trouble following if it is.  

The "EOS sync" mechanism here is meant to cause an EOS `nullptr` to sit on each input queue until all input queues have `nullptr` at their fronts.  That is, when `neos` equals the number of input queues.  Then a single `nullptr` is sent out the output queue.